### PR TITLE
Fixes bullets passing through the environment

### PR DESCRIPTION
**Pull request by PersonCyclical**
1. Fixes bullets passing through materials
2. Damage now occurs when the bullets hit destructibles and entities, and now correctly damages moving objects
3. Fix a bug which meant that bullets spawned in the centre of the screen for one frame
Note that hit_info.gd is no longer used and should be able to be  safely removed

### CODE REVIEW

**Comment by PersonCyclical** [comment]
Oh also with a slight tweak the bullets can be used to damage the player

File: Ammunition/Shared/pellet.gd [hunk 1]
```diff
@@ -1,11 +1,47 @@
...
+	
+	_start_position = global_position;
 func _physics_process(_delta):
-	global_position += Vector2.RIGHT.rotated(rotation) * _pellet_speed
-
-
+	var old_position := global_position;
+	#TODO find best distance to check
+	var position_delta := Vector2.RIGHT.rotated(rotation) * _pellet_speed;
+	var target := global_position + position_delta
```
Comment (WdBros) [inline Ammunition/Shared/pellet.gd]: i'll split this into its own function, something like `_next_frame_position()`

**Review by WdBros — CHANGES_REQUESTED** [review]
hey so this looks good and everything works but i will have to ask you to please clean up the formatting and some of the chunkier blocks of code with something like gaps of white space, specifically in pellet.gd. (please closely refer to our style guide)
once you're done i'll squash the commits when necessary and we'll merge :3

File: Ammunition/Shared/pellet.gd (hunk 1 above)
Comment (PersonCyclical) [reply Ammunition/Shared/pellet.gd]: Unfortunately there was a bug that was in the visual display code which meant that the bullet speed was framerate dependent. Its fixed now but using the _previous_frame_position() will now require storing the previous frames delta value.

**Review by WdBros — APPROVED** [review]
very cool :+1:
i'll merge this, thank u for ur work <3

**Pull request merged**